doc/rst/technotes/dsi.rst

```diff
@@ -954,5 +954,173 @@ implicitly for each ``GlobalDomain`` object and so will be local in any case.
+
+Each method shares these arguments:
+
+* ``myDom:domain`` - the indices used when transferring to/from ``this``
+
+* ``otherClass`` - a class inheriting from ``BaseArr`` that represents the
+  other array in the assignment
+
+* ``otherDom:domain`` - the indices used when transferring to/from ``otherClass``
+
+These arguments have the following guarantees:
+
+* ``this.rank == myDom.rank``
+
+* ``otherClass.rank == otherDom.rank``
+
+* ``myDom.size == otherDom.size``
+
+.. note::
+
+  ``myDom`` and ``otherDom`` are not required to be distributed, even if the
+  corresponding arrays are distributed. They simply represent the indices
+  involved in the transfer.
+
+Each method returns a ``bool`` indicating whether the assignment occurred. This
+allows domain maps to default to the Chapel implementation of array assignment
+if they are incapable or unwilling to perform the assignment themselves.
+
+Known vs. Any methods
+---------------------
+
+There are two kinds of transfers accounted for in this interface.
+
+The first kind of transfer is a ``Known`` transfer. These are transfers that
+the implementer believes to be optimal. Typically this means that the
+implementer knows the type and internals of the other domain map involved in
+the transfer.
+
+The second kind of transfer is an ``Any`` transfer. These are transfers to/from
+domain maps whose type and internals are **not** known. These transfers rely on
+a common interface, which may involve unnecessary overhead compared to a
+``Known`` transfer. For example, it is likely that many domain maps will
+implement ``Known`` transfers to/from Chapel's default rectangular arrays. It
+is also likely that many domain maps will be implemented using default
+rectangular arrays. Knowing this, implementers of ``Any`` transfers can
+implement in terms of ``Known`` transfers to/from their own default rectangular
+arrays as a common interface.
+
+This interface recognizes these kinds of transfers because otherwise it is
+difficult to determine which distribution implements the best transfer.
+
+For example, the author of ``PopularDist`` could implement ``Any`` methods for
+better performance with unknown third-party domain maps. The author of
+``NewDist`` could implement ``Known`` methods to ``PopularDist`` due to its
+popularity. By doing so the author of ``NewDist`` has informed Chapel that it
+knows better than ``PopularDist`` and a more optimal transfer occurs.
+
+Chapel will call ``Known`` methods before ``Any`` methods if they can be
+resolved. A recommended convention is to implement ``Known`` methods with a
+where-clause that constrains the type of the ``otherClass`` argument. This
+prevents the ``Known`` method from being called with unknown types.
+
+When Does a Bulk-Transfer Occur?
+--------------------------------
+
+There are a number of conditions that need to be true for a bulk-transfer to
+occur:
+
+* The config-param ``useBulkTransfer`` must be true
```

Comment:
Can you link to the documentation for this param?

Reply:
I was thinking this document could serve as the documentation for this param. Would a sentence like "the useBulkTransfer config param is true by default" suffice? It's located in ``ChapelArray``, but that doesn't seem like the right place to document it.

Reply:
That seems reasonable, since this is why it was defined.  If that's the intention, maybe add an example of how to set it and/or give it its own section?  Something that makes it feel like you've gotten enough information and don't need to go looking for more.